cdap-app-templates/cdap-etl/cdap-etl-core/src/main/java/io/cdap/cdap/etl/common/ConnectionMacroEvaluator.java

```diff
@@ -60,11 +62,11 @@ String evaluateMacro(String macroFunction,
       throw new InvalidMacroException("Macro '" + FUNCTION_NAME + "' should have exactly 1 arguments");
     }
 
+    String path = URLEncoder.encode(String.format("v1/contexts/%s/connections/%s",
```

Comment:
would it be better to put the encoding logic in ServiceDiscoverer ?

Reply:
No, because some path will contain query param which includes "?" and "&", and we do not want to encode these characters.